src/devicewrapper.h

```diff
@@ -0,0 +1,249 @@
+typedef void(*dd_mode)(int mode, pDevDesc dd);
+typedef void(*dd_newPage)(const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_polygon)(int n, double *x, double *y, const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_polyline)(int n, double *x, double *y, const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_rect)(double x0, double y0, double x1, double y1, const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_path)(double *x, double *y, int npoly, int *nper, Rboolean winding, const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_raster)(unsigned int *raster, int w, int h, double x, double y, double width, double height, double rot, Rboolean interpolate, const pGEcontext gc, pDevDesc dd);
+typedef SEXP(*dd_cap)(pDevDesc dd);
+typedef void(*dd_size)(double *left, double *right, double *bottom, double *top, pDevDesc dd);
+typedef double(*dd_strWidth)(const char *str, const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_text)(double x, double y, const char *str, double rot, double hadj, const pGEcontext gc, pDevDesc dd);
+typedef void(*dd_onExit)(pDevDesc dd);
+typedef SEXP(*dd_getEvent)(SEXP, const char *);
+typedef Rboolean(*dd_newFrameConfirm)(pDevDesc dd);
+typedef void(*dd_eventHelper)(pDevDesc dd, int code);
+typedef int(*dd_holdflush)(pDevDesc dd, int level);
+typedef void(*dd_textUTF8)(double x, double y, const char *str, double rot, double hadj, const pGEcontext gc, pDevDesc dd);
+typedef double(*dd_strWidthUTF8)(const char *str, const pGEcontext gc, pDevDesc dd);
+
+#define RHOST_DEV_WRAPPER_DECL_GET(member_type, member) \
+            member_type get_##member(); \
+
+#define RHOST_DEV_WRAPPER_DECL_SET(member_type, member) \
+            void set_##member(member_type); \
+
+#define RHOST_DEV_WRAPPER_DECL_GET_ARRAY(member_type, member) \
+            member_type get_##member(int); \
+
+#define RHOST_DEV_WRAPPER_DECL_SET_ARRAY(member_type, member) \
+            void set_##member(int, member_type); \
+
+#define RHOST_DEV_WRAPPER_DECL_GETSET(member_type, member) \
+RHOST_DEV_WRAPPER_DECL_GET(member_type, member) \
+RHOST_DEV_WRAPPER_DECL_SET(member_type, member) \
+
+#define RHOST_DEV_WRAPPER_DECL_GETSET_ARRAY(member_type, member) \
+RHOST_DEV_WRAPPER_DECL_GET_ARRAY(member_type, member) \
+RHOST_DEV_WRAPPER_DECL_SET_ARRAY(member_type, member) \
+
+#define RHOST_DEV_DEREF(ver, member) dd_##ver->##member
+#define RHOST_DEV_DEREF_ARRAY(ver, member) dd_##ver->##member[index]
+
+#define RHOST_DEV_GET_CASE(ver, member) \
+    case ver: return RHOST_DEV_DEREF(ver, member); \
+
+#define RHOST_DEV_GET_CASE_ARRAY(ver, member) \
+    case ver: return RHOST_DEV_DEREF_ARRAY(ver, member); \
+
+#define RHOST_DEV_SET_CASE(ver, member) \
+    case ver: RHOST_DEV_DEREF(ver, member) = value; break; \
+
+#define RHOST_DEV_SET_CASE_ARRAY(ver, member) \
+    case ver: RHOST_DEV_DEREF_ARRAY(ver, member) = value; break; \
+
+#define RHOST_DEV_GET_CASE_COMMON(member) \
+    RHOST_DEV_GET_CASE(R_33_GE_version, member) \
```

Comment:
>R_33_GE_version [](start = 23, length = 15)

since you're using 11 and 12 everywhere else in devicewrapper.h/.cpp, it's probably less confusing if you use 11 and 12 here too, instead of R_33, R_34...